### PR TITLE
issue: 1512054 Enable direct mlx5 processing on VMs

### PR DESCRIPTION
**Pull request by igor-ivanov**
This change includes fix for ilegal instruction failure.

Signed-off-by: Igor Ivanov <igor.ivanov.va@gmail.com>

### CODE REVIEW

File: src/vma/dev/qp_mgr_eth_mlx5.cpp [hunk 1]
```diff
@@ -224,25 +224,40 @@ inline void qp_mgr_eth_mlx5::set_signal_in_next_send_wqe()
...
 //! Copying src to BlueFlame register buffer by Write Combining cnt WQEBBs
-static inline void copy_bf(uint64_t *bf_dst, uint64_t *src, int cnt)
+inline void qp_mgr_eth_mlx5::copy_bf(uint64_t *bf_dst, uint64_t *src, int cnt)
 {
 	// Avoid using memcpy() to copy to BlueFlame page, since memcpy()
 	// implementations may use move-string-buffer assembler instructions,
 	// which do not guarantee order of copying.
-	while (cnt--) {
-		COPY_64B_NT(bf_dst, src);
+	if (m_b_is_hypervisor) {
```
Comment (liranoz12) [inline src/vma/dev/qp_mgr_eth_mlx5.cpp]: As far as i remember, there were no issues over Windows hyperv,
Assuming I remember correctly, there is no reason to automatically disable it for each VM.

**Comment by igor-ivanov** [comment]
updated with review changes

File: src/vma/dev/qp_mgr_eth_mlx5.cpp [hunk 2]
```diff
@@ -75,6 +75,80 @@ static inline uint64_t align_to_WQEBB_up(uint64_t val)
...
+	,m_sq_wqe_counter(0)
+	,m_dm_enabled(0)
+{
+	if (call_configure && configure(p_rx_comp_event_channel)) {
+		throw_vma_exception("failed creating qp_mgr_eth");
+	}
+
+	copy_bf = &copy_bf_asm;
+	copy_bf2 = &copy_bf2_asm;
+	if (mce_sys_var::HYPER_KVM == safe_mce_sys().hypervisor) {
```
Comment (liranoz12) [inline src/vma/dev/qp_mgr_eth_mlx5.cpp]: This verification is to "general" - we must understand how mlx5 detect this limitation...

**Comment by liranoz12** [comment]
bot:retest

**Comment by liranoz12** [comment]
bot:retest

**Comment by igor-ivanov** [comment]
See updated version

**Comment by liranoz12** [comment]
@igor-ivanov, 
i see significant degradation while memic is used.
please check it.

**Pull request merged**